### PR TITLE
Adds a MultiValueInput component

### PR DESCRIPTION
**Pull request by mpeels**
## Description

Adds a MultiValueEntry component for use during the extended patient create. Styling for the DataTable and left to right form inputs are out of scope and will be addressed in a future PR.

Also uses the existing AlertBanner component which has some styling issues.

## Tickets

* [CNFT1-2882](https://cdc-nbs.atlassian.net/browse/CNFT1-2882)

## Working example
example

## Error 
Screenshot 2024-08-20 at 11 34 29 AM

## Tooltips
Screenshot 2024-08-20 at 11 32 28 AM

Screenshot 2024-08-20 at 11 32 31 AM

Screenshot 2024-08-20 at 11 32 32 AM

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

[CNFT1-2882]: https://cdc-nbs.atlassian.net/browse/CNFT1-2882?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/modernization-ui/src/design-system/entry/multi-value/MultiValueEntry.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+    useEffect(() => {
+        onChange(data);
+    }, [data]);
+
+    useEffect(() => {
+        isDirty(form.formState.isDirty);
+    }, [form.formState.isDirty]);
+
+    const handleSubmit = () => {
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/design-system/entry/multi-value/MultiValueEntry.tsx:44]: suggestion:  The call to form.values can be replaced by using the form's `handleSubmit` function.  It should accept a function that accepts `V`.

File: apps/modernization-ui/src/design-system/entry/multi-value/MultiValueEntry.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    errors,
+    onChange,
+    isDirty,
+    formRenderer,
+    viewRenderer
+}: Props<V>) => {
+    const form = useForm<V>({ mode: 'onBlur', defaultValues });
+    const [data, setData] = useState<V[]>([]);
+    const [state, setState] = useState<'view' | 'edit' | 'new'>('new');
+    const [activeIndex, setActiveIndex] = useState<number | undefined>();
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/design-system/entry/multi-value/MultiValueEntry.tsx:34]: suggestion:  it seems like this state is related, should it be one object?  Perhaps a reducer?  Something similar is done with [useTableActionState](https://github.com/CDCgov/NEDSS-Modernization/blob/main/apps/modernization-ui/src/table-action/useTableActionState.ts)

**Review by adamloup-enquizit — COMMENTED** [review]
This is cool, I think it's very close if you would like to polish some

File: apps/modernization-ui/src/components/button/Button.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import classNames from 'classnames';
 type Props = {
     className?: string;
     icon?: ReactNode;
-    children?: string;
+    children?: string | ReactNode;
```
Comment (mpeels) [inline apps/modernization-ui/src/components/button/Button.tsx:9]: Issue: string here is redundant

**Review by adamloup-enquizit — APPROVED** [review]
👍 Nice!

**Review by hclarkEnq — APPROVED** [review]

**Pull request merged**